pkg/pillar/types/global.go

```diff
@@ -114,6 +117,165 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
+	log.Debugf("Adding int item %s", key)
+	if defaultInt < min || defaultInt > max {
+		log.Fatalf("Adding int item %s failed. Value does not meet given min/max criteria", key)
+	}
+	configItem := ConfigItemSpec{
+		ItemType:   CONFIGITEM_TYPE_INT,
+		Key:        key,
+		intDefault: defaultInt,
+		intMin:     min,
+		intMax:     max,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added int item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddBoolItem(key string, defaultBool bool) {
+	log.Debugf("Adding bool item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:    CONFIGITEM_TYPE_BOOL,
+		Key:         key,
+		boolDefault: defaultBool,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added bool item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddStringItem(key string, defaultString string) {
+	log.Debugf("Adding string item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:      CONFIGITEM_TYPE_STRING,
+		Key:           key,
+		stringDefault: defaultString,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added string item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddTriStateItem(key string, defaultTriState TriState) {
+	log.Debugf("Adding tristate item %s", key)
+	configItem := ConfigItemSpec{
+		Key:             key,
+		ItemType:        CONFIGITEM_TYPE_TRISTATE,
+		triStateDefault: defaultTriState,
+	}
+	configMap.itemMap[key] = configItem
+	log.Debugf("Added tristate item %s", key)
+}
+
+func (configMap *ConfigItemSpecMap) AddAgentSettingItem(key string, defaultSetting string) {
+	log.Debugf("Adding bool item %s", key)
+	configItem := ConfigItemSpec{
+		ItemType:    CONFIGITEM_TYPE_STRING,
+		Key:         key,
+		settingDefault: defaultSetting,
+	}
+	configMap.agentSettings[key] = configItem
+	log.Debugf("Added bool item %s", key)
+}
+
+type GlobalConfigItem struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intValue	    uint32
+	stringValue		string
+	settingValue	string
+	boolValue	    bool
+	triStateValue	TriState
+}
+
+type GlobalConfigItemMap struct {
+	globalSettings map[string]GlobalConfigItem
+	agentSettings  map[string]GlobalConfigItem
+}
+
+func (specMap *ConfigItemSpecMap) parseItem(configMap *GlobalConfigItemMap, item config.ConfigItem) {
+	key := item.Key
+	if _, ok := specMap.itemMap[key]; ok {
+		globalConfigItem := GlobalConfigItem {
+			Key:	key,
+		}
+		if specMap.itemMap[key].ItemType == ConfigItemType(CONFIGITEM_TYPE_INT) {
+			i64, err := strconv.ParseUint(item.Value, 10, 32)
+			if err == nil {
+				globalConfigItem.intValue = uint32(i64)
+			}
+			if globalConfigItem.intValue > specMap.itemMap[key].intMax {
+				globalConfigItem.intValue = specMap.itemMap[key].intMax
+			} else if globalConfigItem.intValue < specMap.itemMap[key].intMin {
+				globalConfigItem.intValue = specMap.itemMap[key].intMin
+			}
+		} else if specMap.itemMap[key].ItemType == ConfigItemType(CONFIGITEM_TYPE_TRISTATE) {
```

Comment:
ConfigItemType(CONFIGITEM_TYPE_TRISTATE)  ==> No need for this typecase if you define the const using appropriate type